js/post.js

```diff
@@ -0,0 +1,193 @@
+import paramenterManagement from "../utils/utils.mjs";
+import postManagement from "../modules/posts.mjs";
+
+const API_BASE_URL = "https://api.noroff.dev/api/v1";
+
+let postContainer = document.querySelector(".postContainer");
+let updateContainer = document.querySelector(".updateContainer");
+
+/**
+ * Initialize the functions for the DOM items
+ * @function
+ */
+function initializeFunctionality() {
+  document
+    .getElementById("logout-pst")
+    .addEventListener("click", logout, false);
+  document
+    .getElementById("update-btn")
+    .addEventListener("click", displayUpdate, false);
+  document
+    .getElementById("delete-btn")
+    .addEventListener("click", deletepst, false);
+}
+
+/**
+ * The logout function should be moved to utils. This is a temporary function that serves the same
+ * purpose as logout in feed. This is something to improve.
+ * @function
+ */
+function logout() {
+  localStorage.clear();
+  window.location.href = "index.html";
+}
+
+/**
+ * get the post of the post pressed from feed. Id can be read from URI
+ * @function
+ */
+async function getpst() {
+  updateContainer.innerHTML = "";
+  return postManagement.getPost(paramenterManagement.getParameter("id"));
+}
+
+/**
+ * delete the post of the post pressed from feed. Id can be read from URI
+ * @function. Timeout to wait for api
+ */
+async function deletepst() {
+  postManagement.deletePost(paramenterManagement.getParameter("id"));
+  setTimeout(displayData, 1000);
+}
+
+/**
+ * update the post of the post pressed from feed. The values are read after pressing the function to display them
+ * under called displayUpdate(). Timeout to wait for api
+ * @function
+ */
+async function updatepst() {
+  postManagement.updatePost(
+    paramenterManagement.getParameter("id"),
+    document.getElementById("update-title").value,
+    document.getElementById("update-body").value,
+    document.getElementById("update-tags").value,
+    "https://images.pexels.com/photos/5929944/pexels-photo-5929944.jpeg?auto=compress&cs=tinysrgb&w=1260&h=750&dpr=1"
+  );
+  setTimeout(displayData, 1000);
+}
+
+/**
+ * update the post of the post pressed from feed. This displays where you can edit the item
+ * @function
+ */
+async function displayUpdate() {
+  updateContainer.innerHTML = `
```

Comment:
This works but could be improved. The user flow could be smoother. When I click to update, I get a form in return with empty fields. 

A better way to think about this is to make the updateContainer a modal and show the post to be updated. This is better UX. The same applies to the delete functionality.